portalnet/src/overlay_service.rs

```diff
@@ -1671,9 +1706,23 @@ where
         let validated_content: Vec<(TContentKey, Vec<u8>)> = join_all(handles)
             .await
             .into_iter()
+            .enumerate()
             // Whether the spawn fails or the content fails validation, we don't want it:
-            .filter_map(|content| content.unwrap_or(None))
+            .filter_map(|(index, content)| content.unwrap_or_else(|err| {
```

Comment:
Looks like we already log when content fails validation & the reason on line 1664? What exactly are the errors you want to explicitly distinguish between here?

Reply:
That isn't catching if the thread fails to be created. Or if a panic happens

Reply:
Panics are just being ignored if they happen in this thread and aren't handled

Reply:
The unwrap_or was unwarping the result of if the thread successfully ran or if it failed. So we weren't handling the errors it was returning.